UnityProject/Assets/BulletMineOnHit.cs

```diff
@@ -0,0 +1,41 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using Mirror;
+
+public class BulletMineOnHit : MonoBehaviour
+{
+	//Script does not work on asteroids but mines AsteroidStation ore
+	public void BulletHitInteract(Collision2D coll, Vector2 Direction)
```

Comment:
Is this called anywhere? I'm not that familiar with bullet logic so correct me if I'm wrong

Reply:
Tried to do logic of breaking ore walls from Bullet not from TilemapDamage. I think it is not good for perfomance to place mining behaviour in TilemapDamage, cuz there are a lot of guns but will be a few mining ones. So I made a script and put a check in BulletKinetic when it Collides. It worked fine with AstroidStation ore walls, as you can see on the gif, but it did not work with actual generated asteroids.